src/Enclave.edl

```diff
@@ -103,6 +103,18 @@ enclave {
          */
         public int occlum_ecall_kill(int pid, int sig);
 
+        /*
+         * Free the userspace of LibOS.
+         *
+         * Userspace VM manager is a global variable, and it will not be freed until exit.
+         * Thus, freeing the userspace VM needs a ECALL before destroying the enclave.
+         *
+         * @retval On success, return 0. On error, return -errno.
+         *
+         * The possible values of errno are
+         *      EAGAIN - The LibOS is not initialized.
+         */
+        public int occlum_ecall_free_user_space();
```

Comment:
Did not get the point why you need to add this ecall? Can you please do the same thing in the enclave destructor function?

Reply:
User space start address is recorded only in trust part. Thus, I think PAL doesn't know the start address of the userspace. ECALL is a way to know this, or I can try `sgx_get_rsrv_mem_info` since there is only one reserved memory area.